src/libasr/codegen/asr_to_c_cpp.h

```diff
@@ -647,6 +647,23 @@ R"(#include <stdio.h>
                 src = "(bool)(" + src + ")";
                 break;
             }
+            case (ASR::cast_kindType::LogicalToReal) : {
+                src = "(float)(" + src + ")";
```

Comment:
Depends on the kind of destination `Real_t` type. `float` should be used for kind `4` and `double` should be used for kind `8`.

Reply:
done! thanks!